portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.Marker;
+import com.google.sps.utilities.InputCleaner;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+import java.util.Optional;
+import java.lang.String;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+
+/** Handles fetching and saving markers data. */
+@WebServlet("/markers")
+public class MarkerServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+        Query query = new Query("Marker");
+        PreparedQuery results = datastore.prepare(query);
+
+        ArrayList<Marker> markers = new ArrayList<Marker>();
+
+        for (Entity entity : results.asIterable()) {
+            
+            Marker marker = createMarker(entity);
+
+            markers.add(marker);
+            
+        }
+
+        response.setCharacterEncoding("UTF-8");
+        response.setContentType("application/json;");
+        response.getWriter().println(convertListToJson(markers));
+
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        //use UTF-8 encoding to support different languages
+        try {
+            request.setCharacterEncoding("UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            System.out.println("Encoding failed.");
```

Comment:
Make this error message more informative like: "UTF-8 encoding is unsupported by your browser"